pkg/core/bundle_unpack.go

```diff
@@ -60,72 +164,305 @@ type errorHit struct {
 	file  string
 }
 
-func unpackDataFiles(ctx context.Context, bundle *Bundle, file string) error {
-	ls := bundle.BundleDescriptor.LeafSize
-	fs, err := cafs.New(
-		cafs.LeafSize(ls),
-		cafs.LeafTruncation(bundle.BundleDescriptor.Version < 1),
-		cafs.Backend(bundle.BlobStore),
-		cafs.ReaderConcurrentChunkWrites(bundle.concurrentFileDownloads/fileDownloadsPerConcurrentChunks),
-	)
+type downloadBundleChans struct {
+	error              chan<- errorHit
+	done               <-chan struct{}
+	doneOk             chan<- struct{}
+	concurrencyControl <-chan struct{}
+}
+
+func downloadBundleEntrySyncMaybeOverwrite(ctx context.Context, bundleEntry model.BundleEntry,
+	bundle *Bundle,
+	fs cafs.Fs,
+	overwrite bool) error {
+	bundle.l.Info("starting bundle entry download",
+		zap.String("name", bundleEntry.NameWithPath))
+	key, err := cafs.KeyFromString(bundleEntry.Hash)
+	if err != nil {
+		return err
+	}
+	reader, err := fs.Get(ctx, key)
+	if err != nil {
+		return err
+	}
+	var putParameterOverwrite bool
+	if overwrite {
+		err = bundle.ConsumableStore.Delete(ctx, bundleEntry.NameWithPath)
```

Comment:
all of the internal api in this file is kinda of a disaster area:  what's convenient to debug, isn't always convenient to maintain.

using `Delete` and `Put` calls rather than calling `Put` with a `storage.Overwrite` parameter is particularly egregious, however:  the `storage.Overwrite` parameter does something different in CI (linux) than local (mac), and this needs to be tested and fixed as a separate issue.